### PR TITLE
Agent ID Cards

### PR DESCRIPTION
**Pull request by Elijahrane**
## About the PR 
Classic traitor item that will go well with identity.
Preemptively: No, I am not adding a doafter.

**Screenshots**

https://user-images.githubusercontent.com/60792108/157344795-6a4f406e-8c88-4bee-a696-fb57af68bd04.mp4

**Changelog**

:cl: Rane
- add: Added Agent ID cards to the uplink, a traitor item that allows you to copy accesses from other cards and change the name and job title to whatever you want.

### CODE REVIEW

**Comment by CrudeWax** [comment]
Does the trim also change depending on the job?

**Comment by Elijahrane** [comment]
No

File: Content.Server/Access/Systems/AgentIDCardSystem.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+            _cardSystem.TryChangeJobTitle(uid, args.Job);
+        }
+
+        private void OnNameChanged(EntityUid uid, AgentIDCardComponent comp, AgentIDCardNameChangedMessage args)
+        {
+            if (!TryComp<IdCardComponent>(uid, out var idCard))
+                return;
+
+            _cardSystem.TryChangeFullName(uid, args.Name);
```
Comment (mirrorcult) [inline Content.Server/Access/Systems/AgentIDCardSystem.cs:63]: pass in the id card comp into the function

File: Content.Server/Access/Systems/AgentIDCardSystem.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            }
+            _popupSystem.PopupEntity(Loc.GetString("agent-id-new", ("number", addedLength), ("card", args.Target)), args.Target.Value, Filter.Pvs(args.User));
+        }
+
+        private void OnJobChanged(EntityUid uid, AgentIDCardComponent comp, AgentIDCardJobChangedMessage args)
+        {
+            if (!TryComp<IdCardComponent>(uid, out var idCard))
+                return;
+
+            _cardSystem.TryChangeJobTitle(uid, args.Job);
```
Comment (mirrorcult) [inline Content.Server/Access/Systems/AgentIDCardSystem.cs:55]: pass in the id card comp

File: Content.Client/Access/UI/AgentIDCardWindow.xaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<DefaultWindow xmlns="https://spacestation14.io"
+            xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+            Title="Agent ID Card">
```
Comment (mirrorcult) [inline Content.Client/Access/UI/AgentIDCardWindow.xaml:3]: loc

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**